fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java

```diff
@@ -1898,7 +1902,10 @@ public Pair<List<TResultBatch>, Status> executeStmtWithExecPlan(ConnectContext c
         try {
             UUID uuid = context.getQueryId();
             context.setExecutionId(UUIDUtil.toTUniqueId(uuid));
-            coord = new Coordinator(context, plan.getFragments(), plan.getScanNodes(), plan.getDescTbl().toThrift());
+
+            ICoordinator.Factory coordFactory = new Coordinator.Factory();
```

Comment:
Could use static Factory method, then we could only use one line code to create the new Coordinator.

Reply:
In the next PR, we may introduce a new scheduler against the current coordinator. By `Factory` interface, we could just create  the different `Factory` derived class according to a session variable to indicate whether use the new scheduler.

Reply:
Extract `getCoordinatorFactory()`, and then we could also use one line to create the new Coordinator. What's best, we could only modify the method `getCoordinatorFactory`, when we want to switch different `Coordinator`s.
```java
    private ICoordinator.Factory getCoordinatorFactory() {
        return new Coordinator.Factory();
    }

```